src/tests.rs

```diff
@@ -76,12 +81,10 @@ impl InstructionReader for VirtualFileSystem {
                         contents: if rbxmx {
                             let tree = rbx_xml::from_str_default(&contents_string)
                                 .expect("couldn't decode encoded xml");
-                            let child_id = tree
-                                .get_instance(tree.get_root_id())
-                                .unwrap()
-                                .get_children_ids()[0];
-                            let child_instance = tree.get_instance(child_id).unwrap().clone();
-                            VirtualFileContents::Instance((*child_instance).clone())
+                            let child_id =
+                                tree.get_by_ref(tree.root().referent()).unwrap().children()[0];

```

Comment:
`tree.root()` already returns the instance, so there's no need to grab its referent and look it up in the tree again.